sling/nlp/ner/idf.cc

```diff
@@ -0,0 +1,258 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "sling/nlp/ner/idf.h"
+
+#include <math.h>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "sling/base/logging.h"
+#include "sling/base/types.h"
+#include "sling/file/repository.h"
+#include "sling/nlp/document/document.h"
+#include "sling/nlp/document/fingerprinter.h"
+#include "sling/string/numbers.h"
+#include "sling/task/documents.h"
+#include "sling/task/accumulator.h"
+#include "sling/util/unicode.h"
+
+namespace sling {
+namespace nlp {
+
+using namespace task;
+
+// Collect vocabulary from document corpus and output word fingerprints with
+// counts.
+class VocabularyMapper : public DocumentProcessor {
+ public:
+  void Startup(Task *task) override {
+    // Initialize accumulator.
+    accumulator_.Init(output(), 1 << 24);
+
+    // Get normalization flags.
+    normalization_ = ParseNormalization(task->Get("normalization", "cln"));
+
+    // Pruning threshold for skipping short documents.
+    task->Fetch("min_document_length", &min_document_legth_);
```

Comment:
typo: should be min_document_length_

Reply:
Oops. Fixed. At least I am consistent in my typos, aka copy-paste :)